server/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -276,14 +292,19 @@ private boolean isPrimaryKey(RelationInfo relationInfo) {
     }
 
     public Iterable<PgProcTable.Entry> pgProc() {
-        return () -> concat(
+        var buildInAndUserDefinedFunctions = concat(
```

Comment:
I think this `concat` here needs to be put into the `() -> ` block further below to make sure that all subsequent `iterator()` calls from the `Iterable` return the same rows. 
As it is I think this stream would be exhausted after the first `iterator` has been processed.

Some of our BatchIterator implementations can depend on that behavior.

Reply:
yes, makes sense! amended the last commit with this change.